common/build/eslint-plugin-fluid/src/test/example/no-unchecked-record-access/fileWithOnlyRecordAccess.ts

```diff
@@ -0,0 +1,119 @@
+ */
+const someObjWithStaticType: StaticType = { a: "hello", b: "goodbye" };
+someObjWithStaticType.a; // ok: Accessing string property 'a'
+someObjWithStaticType.a.length; // ok: Accessing length of string property 'a'
+someObjWithStaticType["a"]; // ok: Accessing property 'a' using bracket notation
+someObjWithStaticType["a"].length; // ok: Accessing length of string property 'a' using bracket notation
+someObjWithStaticType[a].length; // ok: Accessing length of string property 'a' using bracket notation
+someObjWithStaticType[b].length; // ok: Accessing length of string property 'b' using bracket notation
+const aExpectingStringFromStaticType: string = someObjWithStaticType.a; // ok: Assigning string property to a strict string variable
+const aExpectingStringOrUndefinedFromStaticType: string | undefined = someObjWithStaticType.a; // ok: Assigning string property to a string or undefined variable
+
+const indexedRecordOfStrings: IndexSignatureType = someObjWithStaticType;
+indexedRecordOfStrings.a; // ok: Accessing index property 'a' without requiring a particular result
+indexedRecordOfStrings.a.length; // defect: Accessing length of index property 'a', but 'a' might not be present
+indexedRecordOfStrings["a"]; // ok: Accessing property 'a' using bracket notation
+indexedRecordOfStrings["a"].length; // defect: Accessing length of index property 'a' using bracket notation, but 'a' might not be present
+indexedRecordOfStrings[a].length; // defect: Accessing length of index property 'a' using bracket notation, but 'a' might not be present
+indexedRecordOfStrings[b].length; // defect: Accessing length of index property 'b' using bracket notation, but 'b' might not be present
+indexedRecordOfStrings.a?.length; // ok: Using optional chaining to access length safely handles 'undefined'
+indexedRecordOfStrings.a!.length; // ok: The author says they understand the question raised by check and acknowledge that they have other information expecting that it is actually defined or that they are okay with an exception being raise here if "a" is not present and defined
+indexedRecordOfStrings["a"]?.length; // ok: Using optional chaining to access length using bracket notation safely handles 'undefined'
+indexedRecordOfStrings["a"]!.length; // ok: The author says they understand the question raised by check and acknowledge that they have other information expecting that it is actually defined or that they are okay with an exception being raise here if "a" is not present and defined
+
+if (indexedRecordOfStrings.a) {
+	indexedRecordOfStrings.a.length; // ok: Within a truthy check, 'a' is guaranteed to be defined
+}
+
+if ("a" in indexedRecordOfStrings) {
+	indexedRecordOfStrings.a.length; // ok: Accessing length of property inside an 'in' check, 'a' is guaranteed to be defined
+}
+
+function recordAFnExpectsString(record: IndexSignatureType): string {
+	return record.a; // defect: Returning index property 'a' directly, but 'a' might not be present
+}
```

Comment:
Can you place this with `recordAFnExpectsStringOrUndefined`? (It took me a moment to see that you had both cases covered.)